packages/react-composites/src/composites/common/CaptionsSettingModal.tsx

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import React from 'react';
+import { _CaptionsSettingModal } from '@internal/react-components';
+import { usePropsFor } from '../CallComposite/hooks/usePropsFor';
+
+/** @private */
+export const CaptionsSettingModal = (props: {
+  showCaptionsSettingModal: boolean;
+  onDismissCaptionsSetting: () => void;
+}): JSX.Element => {
+  const captionsSettingModalProps = usePropsFor(_CaptionsSettingModal);
```

Comment:
use (useSelector + useAdapter) instead, so we could avoid potential type leaking from main index